src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -379,7 +381,7 @@ private function createQueryBuilder(string $alias = 'm'): QueryBuilder
         ));
     }
 
-    private function executeQuery(string $sql, array $parameters = [], array $types = [])
+    private function executeQuery(string $sql, array $parameters = [], array $types = []): Result|AbstractionResult|Statement|DriverResult
```

Comment:
This is what I derived from the mocks used in the test suite, but it looks way too complex given we only support 2 major DBAL versions (so I would expect at most an union of 2 classes).

Does someone know the actual return type here, so we can update the mocks in the test suite?

Reply:
I dug a bit the git history of DBAL, it looks like `Result|AbstractionResult|ResultStatement` could work.
I pushed it, let's see.